### PR TITLE
Dynamodb Stack

### PR DESCRIPTION
**Pull request by chrispypatt**

### CODE REVIEW

File: lib/constructs/dynamodb/dynamodb-table.ts [hunk 1]
```diff
@@ -1,29 +1,62 @@
...
-    // Output values
-    new CfnOutput(this, "TableName", {
-      value: table.tableName,
-      exportName: app.logicalPrefixedName("TableName"),
-    });
-    new CfnOutput(this, "TableArn", {
-      value: table.tableArn,
-      exportName: app.logicalPrefixedName("TableArn"),
-    });
+    this.scope = scope;
```
Comment (chrispypatt) [inline lib/constructs/dynamodb/dynamodb-table.ts:34]: I wonder if this is why your stack is deploying twice.

File: lib/constructs/dynamodb/dynamodb-table.ts [hunk 2]
```diff
@@ -1,29 +1,62 @@
...
+    console.log(scope.region);
+    this.createDynamoTables(DYNAMO_TABLES);
+  }
+
+  createDynamoTables(tables: Array<DynamoDbTableDefinition>) {
+    for (const table of  tables) {
+      const { tableName, partitionKey } = table;
+
+      this.createTable(`dev-${tableName}`, partitionKey);
+      this.createTable(`prod-${tableName}`, partitionKey);
```
Comment (chrispypatt) [inline lib/constructs/dynamodb/dynamodb-table.ts]: We should remove this and make the stack aware of its stage

File: lib/constructs/dynamodb/dynamodb-table.ts [hunk 3]
```diff
@@ -1,29 +1,62 @@
...
+    }
+  }
+
+  createTable(tableName: string, partitionKey: string) {
+    try {
+      const table = new Table(this, tableName, {
+        tableName: tableName,
+        billingMode: BillingMode.PAY_PER_REQUEST, // Use on-demand billing mode
+        partitionKey: { name: partitionKey, type: AttributeType.STRING },
```
Comment (chrispypatt) [inline lib/constructs/dynamodb/dynamodb-table.ts:54]: push and business tables also have GSIs on them. Each have a GSI for button ids which we should add for now even if we arent going to for sure use buttons

File: lib/constructs/dynamodb/dynamodb-table.ts [hunk 4]
```diff
@@ -1,29 +1,62 @@
...
+      const { tableName, partitionKey } = table;
+
+      this.createTable(`dev-${tableName}`, partitionKey);
+      this.createTable(`prod-${tableName}`, partitionKey);
+    }
+  }
+
+  createTable(tableName: string, partitionKey: string) {
+    try {
+      const table = new Table(this, tableName, {
```
Comment (chrispypatt) [inline lib/constructs/dynamodb/dynamodb-table.ts:51]: We dont have to now but we may want to consider adding Cloudwatch alarms for some of the metrics associated with our tables

File: lib/constructs/dynamodb/dynamodb-table.ts (hunk 1 above)
Comment (branberry) [reply lib/constructs/dynamodb/dynamodb-table.ts:34]: are you referring to this.scope = scope or the table arn stuff? I tried commenting out the scope stuff in the constructor and just logged out a random string, but it still runs twice. It doesn't seem to give me an error anymore about already existing stacks so it looks like the actual database creation only occurs once now.

Comment (chrispypatt) [reply lib/constructs/dynamodb/dynamodb-table.ts:34]: Yes just the scope part. Good that it’s no longer running cfn twice but still I’m curious what is going on

File: lib/constructs/dynamodb/dynamodb-table.ts (hunk 3 above)
Comment (branberry) [reply lib/constructs/dynamodb/dynamodb-table.ts:54]: gotcha adding these in

**Pull request merged**